arbor/fvm_lowered_cell_impl.hpp

```diff
@@ -648,111 +653,367 @@ fvm_size_type fvm_lowered_cell_impl<Backend>::fvm_intdom(
+    mcable_list cables;
 
-    if (auto *p = any_cast<cell_probe_membrane_voltage>(&paddr)) {
-        const fvm_value_type* src = state_->voltage.data() + location_cv(p, cv_empty);
-        return fvm_probe_info{src};
-    }
-    else if (auto* p = any_cast<cell_probe_total_ionic_current_density>(&paddr)) {
-        const fvm_value_type* src = state_->current_density.data() + location_cv(p, cv_nonempty);
-        return fvm_probe_info{src};
+    for (auto cv: R.D.geometry.cell_cvs(R.cell_idx)) {
+        const double* ptr = R.state->voltage.data()+cv;
+        for (auto cable: R.D.geometry.cables(cv)) {
+            r.raw_handles.push_back(ptr);
+            cables.push_back(cable);
+        }
     }
-    else if (auto* p = any_cast<cell_probe_density_state>(&paddr)) {
-        if (auto data = lookup_mechanism_data(p)) {
-            auto cv = location_cv(p, cv_nonempty);
-            if (auto opt_i = binary_search_index(M.mechanisms.at(p->mechanism).cv, cv)) {
-                return fvm_probe_info{data.state_data+opt_i.value()};
+    r.metadata = std::move(cables);
+    r.shrink_to_fit();
+    return r;
+}
+
+template <typename B>
+fvm_probe_info resolve_probe(const cell_probe_axial_current& p, probe_resolution_data<B>& R) {
+    const fvm_value_type* data = R.state->voltage.data();
+    fvm_voltage_interpolant in = fvm_axial_current(R.cell, R.D, R.cell_idx, p.location);
+
+    return fvm_probe_interpolated{
+        {data+in.proximal_cv, data+in.distal_cv},
+        {in.proximal_coef, in.distal_coef},
+        p.location};
+}
+
+template <typename B>
+fvm_probe_info resolve_probe(const cell_probe_total_ion_current_density& p, probe_resolution_data<B>& R) {
+    return fvm_probe_scalar{
+        {R.state->current_density.data() + R.D.geometry.location_cv(R.cell_idx, p.location, cv_prefer::cv_nonempty)},
+        p.location};
+}
+
+template <typename B>
+fvm_probe_info resolve_probe(const cell_probe_total_ion_current_cell& p, probe_resolution_data<B>& R) {
+    fvm_probe_weighted_multi r;
+
+    for (auto cv: R.D.geometry.cell_cvs(R.cell_idx)) {
+        const double* ptr = R.state->current_density.data()+cv;
+        for (auto cable: R.D.geometry.cables(cv)) {
+            double area = R.cell.embedding().integrate_area(cable); // [µm²]
+            if (area>0) {
+                r.raw_handles.push_back(ptr);
+                r.weight.push_back(0.001*area); // Scale from [µm²·A/m²] to [nA].
+                r.metadata.push_back(cable);
             }
         }
-        return fvm_probe_info{};
     }
-    else if (auto* p = any_cast<cell_probe_point_state>(&paddr)) {
-        if (p->target>=handles.size()) {
-            return fvm_probe_info{};
-        }
+    r.shrink_to_fit();
+    return r;
+}
+
+template <typename B>
+fvm_probe_info resolve_probe(const cell_probe_total_current_cell& p, probe_resolution_data<B>& R) {
+    fvm_probe_membrane_currents r;
+
+    auto cell_cv_ival = R.D.geometry.cell_cv_interval(R.cell_idx);
```

Comment:
`D` for the `fvm_discretiazation` is used consistently through `fvm_layout.cpp` and the rest of `fvm_lowered_cell_impl.hpp`. Using a longer name makes some of these expressions even more unwieldy than they are, and I'm pretty sure anyone with their hands in the code will quickly see how `D` is used.

We can make it longer, but I don't think it will consequently aid readability.